ion-schema/src/main/kotlin/com/amazon/ionschema/reader/internal/IonSchemaReaderV2_0.kt

```diff
@@ -9,20 +12,10 @@ import com.amazon.ionschema.model.ExperimentalIonSchemaModel
 import com.amazon.ionschema.model.NamedTypeDefinition
 import com.amazon.ionschema.model.SchemaDocument
 import com.amazon.ionschema.model.TypeDefinition
-import com.amazon.ionschema.reader.internal.FooterReader
-import com.amazon.ionschema.reader.internal.HeaderReader
-import com.amazon.ionschema.reader.internal.ReadError
-import com.amazon.ionschema.reader.internal.ReaderContext
-import com.amazon.ionschema.reader.internal.TypeReaderV2_0
-import com.amazon.ionschema.reader.internal.isFooter
-import com.amazon.ionschema.reader.internal.isHeader
-import com.amazon.ionschema.reader.internal.isTopLevelOpenContent
-import com.amazon.ionschema.reader.internal.isType
-import com.amazon.ionschema.reader.internal.readCatching
 import com.amazon.ionschema.util.IonSchemaResult
 
 @OptIn(ExperimentalIonSchemaModel::class)
-class IonSchemaReaderV2_0 : IonSchemaReader {
+internal object IonSchemaReaderV2_0 : VersionedIonSchemaReader {
```

Comment:
(Question) Why did we move this to be internal? Which reader is available to access by user?

Reply:
The user doesn't directly access any reader. The actual reader is stateless, so there's no advantage to holding a reference to one. Instead, the user calls `IonSchemaReader.readSchema()` or  `IonSchemaReader.readSchemaOrThrow()` (both of which are static functions). This hides the implementation details a little better and will (eventually) be able to read both 1.0 and 2.0 without the user having to know the version ahead of time.